django/http/response.py

```diff
@@ -429,6 +429,28 @@ def getvalue(self):
         return b''.join(self.streaming_content)
 
 
+class FileResponse(StreamingHttpResponse):
+    """
+    A streaming HTTP response class optimized for files.
+    """
+    block_size = 4096
+
+    @property
+    def streaming_content(self):
+        return map(self.make_bytes, self._iterator)
+
+    @streaming_content.setter
+    def streaming_content(self, value):
+        if hasattr(value, 'read'):
+            self.file_to_stream = value
+            self._iterator = iter(lambda: value.read(self.block_size), '')
+        else:
+            self.file_to_stream = None
+            self._iterator = iter(value)
+        if hasattr(value, 'close'):
```

Comment:
The only problem is, I don't think super() works very well with properties, but I might be wrong.